cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/SpaceDevelopersOperations.java

```diff
@@ -60,111 +46,54 @@ public SpaceDevelopersOperations(DefaultCloudFoundryOperations cfOperations) {
     }
 
     /**
-     * Assign a user as a space developer
+     * Prepares a request for the space id to the cf instance.
+     * The space id is needed for assigning/removing space developers.
+     * @return Mono object which yields the space id upon subscription
+     */
+    public Mono<String> getSpaceId() {
+        return cloudFoundryOperations.getSpaceId();
+    }
+
+    /**
+     * Prepares a request for assigning a space developer to the cf instance.
      *
      * @param username email of user to assign as space developer
-     * @throws CreationException when assignation was not successful
-     * @throws NullPointerException when username is null
-     * @throws IllegalArgumentException when username is empty
+     * @param spaceId the id of the space
+     * @return Mono object which yields the response upon subscription
+     * @throws NullPointerException if any of the arguments are null
      */
-    public void assignSpaceDeveloper(String username) throws CreationException {
-        Log.debug("Assigning a space developer:", username);
-
+    public Mono<AssociateSpaceDeveloperByUsernameResponse> assign(@Nonnull String username, @Nonnull String spaceId) {
```

Comment:
Should we setup the logging within the resulting Mono in this method?

Reply:
Yes, it was done in the service operations, so to be consistent it should be done here also.